### PR TITLE
[Android] DiscoveryMessage and DiscoveryMessageParser Classes for UDP Layer

### LINKED ISSUE #88: [Android] UDP: DiscoveryMessage Data Class and Binary Parser
**Parent Issue:** #59

**Description:**
Create the `DiscoveryMessage` data class and implement the binary message parser for UDP discovery broadcasts. This handles the 9-byte message format as specified in API Contract §3.3.

**Related Requirements:** NET1 (LAN communication)

**Protocol Reference (API Contract Section 3.3):**
| Field | Offset | Size | Description |
|-------|--------|------|-------------|
| Opcode | 0 | 1 byte | `0x00` = DISCOVERY |
| IP Address | 1 | 4 bytes | IPv4 address (network byte order, big-endian) |
| HTTP Port | 5 | 2 bytes | Unsigned, little-endian |
| TCP Port | 7 | 2 bytes | Unsigned, little-endian |

**Tasks:**
- [ ] Create `DiscoveryMessage.java` immutable data class in `network/` package
  - Fields: `String ipAddress`, `int httpPort`, `int tcpPort`
  - Include `@NonNull` annotations where appropriate
  - Follow existing domain model patterns (validation in constructor)
- [ ] Create `DiscoveryMessageParser.java` utility class
  - Validate message length (exactly 9 bytes)
  - Validate opcode (must be `0x00`)
  - Parse IPv4 address from bytes 1-4 (big-endian/network byte order)
  - Parse HTTP port from bytes 5-6 (little-endian, unsigned)
  - Parse TCP port from bytes 7-8 (little-endian, unsigned)
  - Use `ByteBuffer` with appropriate byte order
- [ ] Handle malformed messages gracefully (throw `IllegalArgumentException` with descriptive message)
- [ ] Write comprehensive unit tests for parser
  - Valid message parsing
  - Invalid length rejection
  - Invalid opcode rejection
  - Edge cases (port 0, port 65535, IP 0.0.0.0, IP 255.255.255.255)

**Acceptance Criteria:**
- [ ] `DiscoveryMessage.java` created with immutable fields
- [ ] `DiscoveryMessageParser.java` parses 9-byte binary format correctly
- [ ] Opcode validation (0x00)
- [ ] IPv4 address parsing (big-endian)
- [ ] Port parsing (little-endian, unsigned)
- [ ] Malformed message handling
- [ ] 100% test coverage
- [ ] Checkstyle compliant
- [ ] Javadoc for all public methods

**Technical Notes:**
- Use `ByteBuffer.wrap(bytes)` for parsing
- Set byte order with `buffer.order(ByteOrder.LITTLE_ENDIAN)` for ports after reading IP
- For unsigned short: `buffer.getShort() & 0xFFFF`
- IPv4 format: `String.format("%d.%d.%d.%d", b1 & 0xFF, b2 & 0xFF, b3 & 0xFF, b4 & 0xFF)`

**Dependencies:** None - can start immediately

### PR DESCRIPTION
**Pull request by WillStephenn**
This pull request introduces two new classes and their unit test counterparts:

- `DiscoveryMessage.java` is an immutable object containing the contents of the UDP discovery message, forming part of the initial pairing process. The contents include: IP Address, HTTP Port and TCP Port, with input validation on the constructor. If input validation fails, an IllegalArgumentException is thrown.
- `DiscoveryMessageParser.java` provides a static method `parse` for parsing a "Discovery Message" byte array into it's constituent parts (Opcode, IP Address, HTTP Port and TCP Port). Message length (9 Bytes) and Opcode (0x00) are validated - else an IllegalArgumentException is thrown. The `parse` method returns a valid and populated DiscoveryMessage object. 

Closes #88 
Contributes to parent Issue #59

### CODE REVIEW

File: android/app/src/main/java/com/manuscripta/student/network/DiscoveryMessage.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * @param httpPort The HTTP port of the device.
+     * @param tcpPort The TCP port of the device.
+     */
+    public DiscoveryMessage(@NonNull String ipAddress,
+                            int httpPort,
+                            int tcpPort){
+        // Constructor validation
+        if (ipAddress == null){
+            throw new IllegalArgumentException("IP Address cannot be Null");
+        }
```
[the remark this answers is not in the document]
Comment (WillStephenn) [reply android/app/src/main/java/com/manuscripta/student/network/DiscoveryMessage.java]: NonNull tags only produce IDE warnings in Java
